script/tool/test/update_dependency_command_test.dart

```diff
@@ -584,4 +609,197 @@ dev_dependencies:
+        '--version',
+        '8.8.8',
+      ]);
+
+      expect(
+        output,
+        containsAllInOrder(<Matcher>[
+          contains('SKIPPING: Example app does not run on Android.'),
+        ]),
+      );
+    });
+
+    test(
+        'throws if wrapper does not have distribution URL with expected format',
+        () async {
+      final RepositoryPackage package = createFakePlugin(
+          'fake_plugin', packagesDir, extraFiles: <String>[
+        'example/android/app/gradle/wrapper/gradle-wrapper.properties'
+      ]);
+
+      final File gradleWrapperPropertiesFile = package.directory
+          .childDirectory('example')
+          .childDirectory('android')
+          .childDirectory('app')
+          .childDirectory('gradle')
+          .childDirectory('wrapper')
+          .childFile('gradle-wrapper.properties');
+
+      gradleWrapperPropertiesFile.writeAsStringSync('''
+How is it even possible that I didn't specify a Gradle distribution?
+''');
+
+      Error? commandError;
+      final List<String> output = await runCapturingPrint(runner, <String>[
+        'update-dependency',
+        '--packages',
+        package.displayName,
+        '--android-dependency',
+        'gradle',
+        '--version',
+        '8.8.8',
+      ], errorHandler: (Error e) {
+        commandError = e;
+      });
+
+      expect(commandError, isA<ToolExit>());
+      expect(
+        output,
+        containsAllInOrder(<Matcher>[
+          contains(
+              'Failed to update Gradle version in the example app for ${package.displayName}'),
+        ]),
+      );
+    });
+
+    test('succeeds if example app has android/app/gradle directory structure',
+        () async {
+      final RepositoryPackage package = createFakePlugin(
+          'fake_plugin', packagesDir, extraFiles: <String>[
+        'example/android/app/gradle/wrapper/gradle-wrapper.properties'
+      ]);
+      const String newGradleVersion = '8.8.8';
+
+      final File gradleWrapperPropertiesFile = package.directory
+          .childDirectory('example')
+          .childDirectory('android')
+          .childDirectory('app')
+          .childDirectory('gradle')
+          .childDirectory('wrapper')
+          .childFile('gradle-wrapper.properties');
+
+      gradleWrapperPropertiesFile.writeAsStringSync('''
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.1-all.zip
+''');
+
+      await runCapturingPrint(runner, <String>[
+        'update-dependency',
+        '--packages',
+        package.displayName,
+        '--android-dependency',
+        'gradle',
+        '--version',
+        newGradleVersion,
+      ]);
+
+      final String updatedGradleWrapperPropertiesContents =
+          gradleWrapperPropertiesFile.readAsStringSync();
+      expect(updatedGradleWrapperPropertiesContents.contains(newGradleVersion),
```

Comment:
This (and the next test) should look for the whole line, to catch the bug with the `\:`. You can make it extra robust by not using double-escaping, but instead do string concatenation with part of it as a raw string, so that the code generating the string and the code matching the output can't just copy/paste the same bug:

```
...updatedGradleWrapperPropertiesContents, contains(r'distributionUrl=https\://services.gradle.org/distributions/'
    'gradle-$newGradleVersion-all.zip`);

Reply:
(There's a matcher for `contains`, so you can use that directly in `expect` rather than the String `contains` + `isTrue`.)